src/main/java/in/projecteka/consentmanager/user/PatientsController.java

```diff
@@ -195,7 +196,10 @@ private OtpMediumType getOtpMedium(UserSignUpEnquiry userSignUpEnquiry) {
                 ? ReactiveSecurityContextHolder.getContext()
                 .map(securityContext -> (Caller) securityContext.getAuthentication().getPrincipal())
                 .map(Caller::getUsername)
-                .flatMap(userName -> userService.updatePassword(request, userName))
+                .flatMap(userName ->
+                                lockedUserService.validateLogin(userName)
+                                .then(Mono.defer(() -> userService.updatePassword(request, userName)))
+                        )
```

Comment:
why can't the .then be outside of flatmap?

Reply:
but then, it won't be able to access userName in then

Reply:
then return username from validateLogin if username is valid. then do flatMap

Reply:
yeah. will do that